code/modules/atmospherics/FEA_gas_mixture.dm

```diff
@@ -124,15 +125,25 @@ What are the archived variables for?
 			// has already been asserted above instead of utilizing get_trace_gas_by_type()
 			var/datum/gas/oxygen_agent_b/trace_gas = src.trace_gas_refs[/datum/gas/oxygen_agent_b/]
 			if(trace_gas?.moles > MINIMUM_REACT_QUANTITY )
-				reaction_rate = min(src.carbon_dioxide*0.75, src.toxins*0.25, trace_gas.moles*0.05)
-				reaction_rate = QUANTIZE(reaction_rate)
+				// nitrogen saturates oxygen beta, preventing it from reacting with plasma.
+				if (src.nitrogen > src.toxins)
+					reaction_rate = min(src.nitrogen / 2, src.oxygen, trace_gas.moles * SLEEPING_GAS_CATALYSIS_RATIO)
+					reaction_rate = QUANTIZE(reaction_rate * 2)
+					src.oxygen -= reaction_rate / 2
+					src.nitrogen -= reaction_rate
+
+					var/datum/gas/sleeping_agent/sleepy = src.get_or_add_trace_gas_by_type(/datum/gas/sleeping_agent)
+					sleepy.moles += reaction_rate
```

Comment:
Logically it generating something like N2O makes sense but from a gameplay perspective I'm not such a big fan of generating something that is purposefully kept in short supply.